_lzbench/cpuid1.h

```diff
@@ -0,0 +1,38 @@
+#ifndef __CPUID1_H__
+#define __CPUID1_H__
+
+//#include <cpuid.h>
+#include <stdint.h>
+
+enum cpuid_requests {
+  CPUID_GETVENDORSTRING,
+  CPUID_GETFEATURES,
+  CPUID_GETTLB,
+  CPUID_GETSERIAL,
+
+  CPUID_EXTENDED = 0x80000000,
+  CPUID_FEATURES,
+  CPUID_BRANDSTRING,
+  CPUID_BRANDSTRINGMORE,
+  CPUID_BRANDSTRINGEND,
+};
+
+/** issue a single request to CPUID. Fits 'intel features', for instance
+ *  note that even if only "eax" and "edx" are of interest, other registers
+ *  will be modified by the operation, so we need to tell the compiler about it.
+ */
+inline void cpuid2(int code, uint32_t *a, uint32_t *d)
+    {
+    asm volatile("cpuid":"=a"(*a),"=d"(*d):"a"(code):"ecx","ebx");
```

Comment:
The assembly code is not compatible. Do we have a more compatible API under the Linux system?
LIKE cpufreq_get_freq_hardware();

Reply:
What exactly is not compatible?
I just have tested it with mingw.
It's the same thing what __cpuid() does.
Here is [example of inline assembly](https://en.m.wikipedia.org/wiki/CPUID#Inline_assembly) usage. [Here, there is explanation](https://softpixel.com/~cwright/programming/simd/cpuid.php), the best one really, and [here](https://wiki.osdev.org/CPUID) another one (I have utilised in code).